### PR TITLE
pyradio: add requests and dnspython modules

### PR DESCRIPTION
**Pull request by j-steinbach**
This commit adds the missing 'requests' and 'dnspython' modules for the
radio-browser functionality, based on the pyradio packaging readme.

See https://github.com/coderholic/pyradio#packaging-pyradio and https://github.com/NixOS/nixpkgs/issues/151428.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

### Important

This is only my second pull request ever. I am still not very comfortable with this, so carefully scrutinize!

### CODE REVIEW

File: pkgs/applications/audio/pyradio/default.nix
Comment (contrun) [inline pkgs/applications/audio/pyradio/default.nix:19]: Built it with `nixpkgs-review pr 151584`. Works great. Thanks. Two small thing. Please add a separate commit to add yourself as a maintainer. Also, it would be better if you can format the code with. https://github.com/nix-community/nixpkgs-fmt

Comment (j-steinbach) [reply pkgs/applications/audio/pyradio/default.nix:19]: Alright, hopefully the formatting works now.

How do I add myself as maintainer? Do I create a commit and add myself to the https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix? And then do I create another commit to add myself as a maintainer to pyradio?

Comment (contrun) [reply pkgs/applications/audio/pyradio/default.nix:19]: Yeah.

Comment (j-steinbach) [reply pkgs/applications/audio/pyradio/default.nix:19]: This has way more overhead than imagined, but maybe I got it right this time.

Comment (contrun) [reply pkgs/applications/audio/pyradio/default.nix:19]: Look good to me. It may take a while for some committer to merge it.

**Review by contrun — APPROVED** [review]

**Review by contrun — APPROVED** [review]

**Comment by c0bw3b** [comment]
Superseded by #155600

**Comment by j-steinbach** [comment]
Obviously I am a noob, but shouldn't the "new" code contain _all_ my approved changes? When I look at https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/applications/audio/pyradio/default.nix (I hope this is the right one), then I am still missing the `sed -i 's/distro = None/distro = NixOS/' pyradio/config`to disable updating and uninstalling pyradio. Also I added myself as maintainer.

**Comment by c0bw3b** [comment]
@j-steinbach right, my bad. I was focused on the "adding requests and dnspython" part.

Could you rebase you PR on current master?

**Comment by j-steinbach** [comment]
I can, but I don't know how. Do you have perhaps a trusted link where it shows how to do that? Afterwards you just have to give me a few days time :)

File: pkgs/applications/audio/pyradio/default.nix
Comment (c0bw3b) [inline pkgs/applications/audio/pyradio/default.nix:20]: Note that not all usage of nixpkgs is on NixOS.
This could feel weird for someone using nix+nixpkgs on another Linux or on Darwin.

Maybe `distro = nixkgs` is more appropriate? Maybe @contrun has an opinion on this.

Comment (contrun) [reply pkgs/applications/audio/pyradio/default.nix:20]: I agree.

**Comment by contrun** [comment]
> I can, but I don't know how. Do you have perhaps a trusted link where it shows how to do that? Afterwards you just have to give me a few days time :)

I am sorry. I completely forgot this PR. I think nixpkgs has too many ongoing PRs that some PRs that are ready to merge can get lost. I, myself don't have commit permission. So this PR has been hang out here for a long time. Sorry for that.

**Comment by c0bw3b** [comment]
> I can, but I don't know how. Do you have perhaps a trusted link where it shows how to do that? Afterwards you just have to give me a few days time :)

This is not an easy scenario if you are not too comfortable with Git usage.

[See this post for an example](https://verdantfox.com/blog/view/how-to-git-rebase-mainmaster-onto-your-feature-branch-even-with-merge-conflicts) on how to rebase your branch onto master and resolving conflicts that may arise after each of you 3 commits.

`origin` may not be the name of your remote, and you may have more than one remote (one for your own nixpkgs fork and one for NixOS/nixpkgs repo). Check with `git remote -v`.

You will also need to rewrite the commit message of your 1st commit because it will only do the patching on `pyradio/config.`
Use [interactive rebase of your local branch](https://www.atlassian.com/git/tutorials/rewriting-history#git-rebase-i) (reword option for the commit that needs it) in order to do that.

File: pkgs/applications/audio/pyradio/default.nix
Comment (c0bw3b) [reply pkgs/applications/audio/pyradio/default.nix:20]: It's also preferable to use `substituteInPlace` over `sed`.
[Here is one example](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/audio/abcde/default.nix#L27-L28) of its usage in another package.

**Comment by j-steinbach** [comment]
Alright, thanks for the links and advice :) I'll try this myself, but I don't think I can sit down and focus on it before the next weekend.

File: pkgs/applications/audio/pyradio/default.nix
Comment (j-steinbach) [reply pkgs/applications/audio/pyradio/default.nix:20]: Good point(s). I don't have any strong opinion on this. I choose `sed`, as it is the one being used in the README. I choose NixOS, as just `nix` is confusing (means `nothing` in german/dutch languages). `nixpkgs` is a nice solution.

**Comment by j-steinbach** [comment]
I don't think I have any time for this any time soon, so I will close this and maybe create a new PR in the future.

**Pull request closed without merging**
